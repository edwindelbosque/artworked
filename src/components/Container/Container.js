import React from 'react';
import './Container.scss';
import Artwork from '../Artwork/Artwork';
import loadingGif from '../../assets/loadingGif.gif';
import { connect } from 'react-redux';

const Container = ({ results, isLoading }) => {
	const card = results.map((result, index) => {
		const { name, artist, releaseYear, artwork, hqArtwork } = result;
		return (
			<Artwork
				key={index}
				name={name}
				artist={artist}
				releaseYear={releaseYear}
				artwork={artwork}
				hqArtwork={hqArtwork}
			/>
		);
	});

	const loader = (
		<img
			rel='preload'
			alt='loading gif'
			as='image'
			src={loadingGif}
			style={{ width: '600px' }}
		/>
	);

	return (
		<>
			{isLoading && loader}
			<article>{card}</article>
		</>
	);
};

const mapStateToProps = ({ results, isLoading }) => ({
	results,
	isLoading
});

export default connect(mapStateToProps)(Container);
