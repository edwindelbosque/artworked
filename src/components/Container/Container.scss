.Container {
	margin-top: 180px;
}
