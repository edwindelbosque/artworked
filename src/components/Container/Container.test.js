import React from 'react';
import { Container, mapStateToProps } from './Container';
import { shallow } from 'enzyme';

describe('Container', () => {
	let wrapper;
	let favorites;
	let results;

	beforeEach(() => {
		results = [{ id: 12 }, { id: 13 }, { id: 15 }];
		favorites = [{}, {}];
		wrapper = shallow(<Container favorites={favorites} results={results} />);
	});

	it('should match Container snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('mockStateToProps', () => {
		it('should match the props from Redux', () => {
			const mockState = {
				results: [{}, {}, {}],
				isLoading: true,
				isFavorites: false,
				favorites: [{}, {}]
			};
			const expected = {
				results: [{}, {}, {}],
				isLoading: true,
				isFavorites: false,
				favorites: [{}, {}]
			};

			const mappedProps = mapStateToProps(mockState);

			expect(mappedProps).toEqual(expected);
		});
	});
});
