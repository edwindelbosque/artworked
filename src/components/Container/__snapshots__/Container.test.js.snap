// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match Container snapshot 1`] = `
<Fragment>
  <article
    className="Container"
  >
    <Connect(ArtworkResult)
      id={12}
      isLoaded={false}
      key="12"
    />
    <Connect(ArtworkResult)
      id={13}
      isLoaded={false}
      key="13"
    />
    <Connect(ArtworkResult)
      id={15}
      isLoaded={false}
      key="15"
    />
  </article>
</Fragment>
`;
